src/ipc/print.ts

```diff
@@ -57,9 +67,29 @@ export default function register(ipcMain: IpcMain): void {
     channel,
     async (
       event: IpcMainInvokeEvent,
-      deviceName = getPreferredPrinterName(event.sender.getPrinters()),
-      paperSource = '',
+      { deviceName, paperSource, copies }: PrintOptions = {},
     ) => {
+      if (typeof deviceName !== 'undefined' && typeof deviceName !== 'string') {
+        throw new TypeError(
+          `deviceName expected to be a string, got: ${inspect(deviceName)}`,
+        )
+      }
+
+      if (
+        typeof paperSource !== 'undefined' &&
+        typeof paperSource !== 'string'
+      ) {
+        throw new TypeError(
+          `paperSource expected to be a string, got: ${inspect(paperSource)}`,
+        )
+      }
+
+      if (typeof copies !== 'undefined' && typeof copies !== 'number') {
```

Comment:
probably here for the integer check.

Reply:
Done. Replaced the manual checks with a Zod schema.